plugins/scaffolder-backend/src/scaffolder/actions/builtin/publish/gitlab.ts

```diff
@@ -149,19 +157,29 @@ export function createPublishGitlabAction(options: {
         id: number;
       };
 
+      const { id: userId } = (await client.Users.current()) as {
+        id: number;
+      };
+
       if (!targetNamespace) {
-        const { id } = (await client.Users.current()) as {
-          id: number;
-        };
-        targetNamespace = id;
+        targetNamespace = userId;
       }
 
-      const { http_url_to_repo } = await client.Projects.create({
+      const { id: projectId, http_url_to_repo } = await client.Projects.create({
         namespace_id: targetNamespace,
         name: repo,
         visibility: repoVisibility,
       });
 
+      if (setUserAsOwner && integrationConfig.config.token) {
```

Comment:
Maybe we can add a comment here above that describes the behaviour noted in the chat below and why the `integration.config` needs to be set etc.

Reply:
Yeah sure, so that others can understand right away why it is needed

Reply:
Just added the comment. Hope it is clear enough.